src/main/java/com/airlines/dao/impl/AirplaneDaoImpl.java

```diff
@@ -44,13 +44,27 @@ public void save(Airplane airplane) {
             ResultSet generatedKeys = insertStatement.getGeneratedKeys();
             if (generatedKeys.next()) {
                 long id = generatedKeys.getLong(1);
-                airplane.setId(id);
+                return Optional.of(copyAttributesAndSetAirplaneId(airplane, id));
+            } else {
+                return Optional.empty();
             }
         } catch (SQLException e) {
-            throw new DaoOperationException("Error saving product: " + airplane, e);
+            throw new DaoOperationException("Couldn't save product ", e);
         }
     }
 
+    private Airplane copyAttributesAndSetAirplaneId(Airplane airplane, Long id) {
+        return Airplane.builder()
+                .withId(id)
+                .withCodeName(airplane.getCodeName())
+                .withModel(airplane.getModel())
+                .withManufactureDate(airplane.getManufactureDate())
+                .withCapacity(airplane.getCapacity())
+                .withFlightRange(airplane.getFlightRange())
+                .withCrew(airplane.getCrew())
+                .build();
+    }
+
```

Comment:
private methods always located after public methods

Reply:
done in https://github.com/Matsukata/airlines-jdbc-app/pull/5/commits/625cab1de0c97028d188cc5a80294e281f46b375